### PR TITLE
ActionSet refactoring & Extra View Removal.

### LINKED ISSUE #3784: [iOS][Card-Height-Rendering] [Card heights changes based on number of actions in ActionsSet]
# Platform

* iOS

# Author or host

TeamSpace iOS app

# Version of SDK

latest-master

# Details

The card height gets changed based on the number of actions in ActionSet although the actions are supposed to be a scrollable view. The change in height is the same as ('number of buttons added' x 'height of a button').
PFA the screenshot and JSON.
Simulator Screen Shot - iPhone X - 2020-02-07 at 10 40 05
[8ActionsJson.txt](https://github.com/microsoft/AdaptiveCards/files/4203172/8ActionsJson.txt)

### PR DESCRIPTION
**Pull request by nesalang**
Fixed ActionSet having extra height.
This is a back port from release/1.2 branch.
## Related Issue
fixed #3784 

## Description
There was an extra view between base elements and actions causing the view's expansion.

## How Verified
Verified that extra view was removed.
Worked with the Teams mobile to confirm it.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3818)

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm [hunk 1]
```diff
@@ -100,27 +116,26 @@ - (UIView *)renderButtons:(ACRView *)rootView
...
+                                       hostConfig:config];
             [childview addArrangedSubview:button];
         } @catch (ACOFallbackException *exception) {
-            handleActionFallbackException(exception,
-                                          superview,
-                                          rootView,
-                                          inputs,
-                                          acoElem,
-                                          config,
+            handleActionFallbackException(exception, superview, rootView, inputs, acoElem, config,
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: if all the other changes were to change the parameters to be arranged vertically, why is this one to make the parameters arranged horizontally?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm [hunk 2]
```diff
@@ -60,32 +68,40 @@ - (UIView *)renderButtons:(ACRView *)rootView
     ACRRegistration *reg = [ACRRegistration getInstance];
     ACOFeatureRegistration *featureReg = [ACOFeatureRegistration getInstance];
 
-    UIView<ACRIContentHoldingView> *childview = nil;
+    UIStackView *childview = [[UIStackView alloc] init];
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: why was the previous design using some of our custom views? what advantages does changing to a UIStackview has?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm [hunk 3]
```diff
@@ -60,32 +68,40 @@ - (UIView *)renderButtons:(ACRView *)rootView
...
+    childview.distribution = UIStackViewDistributionFillProportionally;
     AdaptiveCards::ActionsConfig adaptiveActionConfig = [config getHostConfig] -> GetActions();
-    NSDictionary<NSString *, NSNumber *> *attributes =
-        @{@"spacing" : [NSNumber numberWithInt:adaptiveActionConfig.buttonSpacing],
-          @"distribution" : [NSNumber numberWithInt:UIStackViewDistributionFillProportionally]};
 
     if (ActionsOrientation::Horizontal == adaptiveActionConfig.actionsOrientation) {
-        childview = [[ACRColumnSetView alloc] initWithFrame:CGRectMake(0, 0, superview.frame.size.width, superview.frame.size.height) attributes:attributes];
-        ((ACRColumnSetView *)childview).isActionSet = YES;
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: what other parts of the code use the isActionSet property? is it still a good idea to keep this flag existing for something else?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm [hunk 4]
```diff
@@ -303,7 +303,7 @@ void applyBackgroundImageConstraints(const BackgroundImage *backgroundImagePrope
                 isDeficientInHeight = YES;
             }
 
-            if (isDeficientInWidth and isDeficientInWidth) {
+            if (isDeficientInWidth and isDeficientInHeight) {
```
Comment (almedina-ms) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm]: isn't this change in other PR too?

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm]: yes, it was added by cherry-pick; shouldn't matter though as the conflict can be resolved. 

---
In reply to: [387951953](https://github.com/microsoft/AdaptiveCards/pull/3818#discussion_r387951953) [](ancestors = 387951953)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm (hunk 1 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: this is done by clang-format according to Apple Coding style.

---
In reply to: [387950316](https://github.com/microsoft/AdaptiveCards/pull/3818#discussion_r387950316) [](ancestors = 387950316)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm (hunk 3 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: This is for layoutsubview method to readjust the action set; this task is handled by ACRContentHoldingUIScrollView atm, but the code can be still used if we others use containing view to the custom view. 

---
In reply to: [387951641](https://github.com/microsoft/AdaptiveCards/pull/3818#discussion_r387951641) [](ancestors = 387951641)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm (hunk 2 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: we don't need custom view's extra feature. stock UIStackView is sufficient to get the job done.

---
In reply to: [387950942](https://github.com/microsoft/AdaptiveCards/pull/3818#discussion_r387950942) [](ancestors = 387950942)

**Pull request merged**